### PR TITLE
DAU Pixel

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1157893581871903/1204238605019198

### Description
This PR adds a new DAU pixel for browser specific features. The pixel is sent once a day with the enabled / disabled state of features. Features can use the `BrowserFeatureStateReporterPlugin` to report state to the pixel.

### Steps to test this PR

_Fresh install_
- [x] Launch app
- [x] Verify `m_browser_feature_daily_active_user_d` is sent with `default_browser=0 `as parameter
- [x] Set browser as default during Dax Onboarding
- [x] Verify `m_browser_feature_daily_active_user_d` is sent with `default_browser=1 `as parameter
- [ ] close and re-launch the app
- [ ] Verify `m_browser_feature_daily_active_user_d` is NOT sent
- [ ] Set a different browser as default browser
- [ ] close and re-launch the app
- [ ] Verify `m_browser_feature_daily_active_user_d` is NOT sent

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2999** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2999" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2999?utm_source=stack-comment).

**Review by aitorvs — DISMISSED** [review]
Left a comment about pixel naming.

LGTM otherwise

File: statistics/src/main/java/com/duckduckgo/app/statistics/pixels/DailyActiveUserPixelSender.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import com.duckduckgo.app.statistics.pixels.Pixel.StatisticsPixelName
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import org.threeten.bp.Instant
+import org.threeten.bp.ZoneOffset
+import org.threeten.bp.format.DateTimeFormatter
+
+@ContributesMultibinding(AppScope::class)
+class DailyActiveUserPixelSender @Inject constructor(
```
Comment (aitorvs) [inline statistics/src/main/java/com/duckduckgo/app/statistics/pixels/DailyActiveUserPixelSender.kt:34]: nit: `DailyActiveUserPixelSender` is a bit confusing. DAUs are calculated in ATB not here. 
How about `FeatureRetentionPixelSender` as we're sending this pixel app retention is refreshed?

**Review by aitorvs — APPROVED** [review]
Left couple comments. Other than that LGTM

File: statistics/src/main/java/com/duckduckgo/app/statistics/api/BrowserFeatureStateReporterPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     * to notify is a feature is enabled or not
+     * @return true if the feature is enabled, false if not
+     */
+    fun isFeatureEnabled(): Boolean
+
+    /**
+     * Identifies the feature name that will be sent as parameter in [StatisticsPixelName.BROWSER_DAILY_ACTIVE_FEATURE_STATE]
+     * @return the name of the feature
+     */
+    fun featureName(): String
```
[the remark this answers is not in the document]
Comment (malmstein) [reply statistics/src/main/java/com/duckduckgo/app/statistics/api/BrowserFeatureStateReporterPlugin.kt:37]: I like that, good suggestion!

File: statistics/src/main/java/com/duckduckgo/app/statistics/pixels/DailyActiveUserPixelSender.kt (hunk 1 above)
Comment (malmstein) [reply statistics/src/main/java/com/duckduckgo/app/statistics/pixels/DailyActiveUserPixelSender.kt:34]: fair enough, makes more sense.

**Comment by malmstein** [comment]
Thanks @aitorvs!

**Pull request merged**